datafusion/expr/src/udaf.rs

```diff
@@ -225,10 +225,10 @@ impl AggregateUDF {
 
     /// See [`AggregateUDFImpl::with_beneficial_ordering`] for more details.
     pub fn with_beneficial_ordering(
-        self,
+        self: Arc<Self>,
         beneficial_ordering: bool,
     ) -> Result<Option<AggregateUDF>> {
-        self.inner
+        Arc::clone(&self.inner)
```

Comment:
The code doesn't explain why `AggregateUDFImpl::with_beneficial_ordering` takes an Arc (it could as well take `&self`), but that's for later